### ISSUE TITLE
Model Gap Fix

### ISSUE THREAD

**Issue by Madis0** [Fabulously-Optimized/fabulously-optimized#473]
### CurseForge link

https://www.curseforge.com/minecraft/mc-mods/model-gap-fix

### CurseForge Mod Distribution

Allowed

### Modrinth link

https://modrinth.com/mod/modelfix

### Source/other link

https://github.com/MehVahdJukaar/modefix-multi

### Mod file size

29.94 KiB

### License

(any other license)

### What it does

#47 but better

### Why should it be in the modpack

Fixes the same bug without lag, also applies to blocks (while the current mod only applies to items)

### Why shouldn't it be in the modpack

Somewhat work in progress

**Comment by Julienraptor01** [comment]
> ### CurseForge link
> 
> https://www.curseforge.com/minecraft/mc-mods/model-gap-fix
> 
> ### CurseForge Mod Distribution
> 
> Allowed
> 
> ### Modrinth link
> 
> https://modrinth.com/mod/modelfix
> 
> ### Source/other link
> 
> https://github.com/MehVahdJukaar/modefix-multi
> 
> ### Mod file size
> 
> 29.94 KiB
> 
> ### License
> 
> (any other license)
> 
> ### What it does
> 
> #47 but better
> 
> ### Why should it be in the modpack
> 
> Fixes the same bug without lag, also applies to blocks (while the current mod only applies to items)
> 
> ### Why shouldn't it be in the modpack
> 
> Somewhat work in progress
> 
> ### Additional details
> 
> _No response_

License is now clear, it's GPL-3.0

**Comment by Madis0** [comment]
Got it. If the license allows redistribution, I don't care which license it is 😄

**Comment by Julienraptor01** [comment]
> Got it. If the license allows forks and redistribution, I don't care which license it is 😄

Yep, i'll try to retest it today or tommorow in case this issue affect FO
https://github.com/MehVahdJukaar/modelfix-multi/issues/2
I'll try most popular resource packs on CF
If some other people can try that too and it doesn't have problems i guess it can be included in next FO

**Comment by Timolino456** [comment]
The issue above seems to be fixed with the new 1.5 version they uploaded. At least I don't experience it anymore.

**Comment by Madis0** [comment]
Seems like no conclusion yet, one person reported it improved perf and the other that it harmed :shrug:

**Comment by RaptaG** [comment]
(Apparently) as of version 1.8 of the mod, no gaps can now be seen. Also, if gaps still appear, the is a config now which you can modify to hide the possible gaps.

https://github.com/MehVahdJukaar/modelfix-multi/issues/4#issuecomment-1277494469

**Comment by osfanbuff63** [comment]
From Discord (myself):
> so i've been long-term testing this for a while, and here's the results:
> - certain resourcepacks cause rendering issues, specifically on leaves for me
> - doesn't really have a noticeable gain or loss in performance or looks of textures
> 
> tl;dr i see no reason to add it at the moment, given it doesn't seem to do much that we don't already have

https://discord.com/channels/859124104644788234/1023944283797065808/1037691495806939227

**Comment by Thanos1716** [comment]
The 1.4.1 or 1.4.2 update of Iris seems to have fixed all of the rendering issues that I was having, relating to [this issue](https://github.com/MehVahdJukaar/modelfix-multi/issues/2) (mainly with Continuity builtins and CITs in combination for some reason), and from my experience it does improve the experience, especially with certain 3D resourcepacks:
Vanilla:
![2022-11-22_17 35 00](https://user-images.githubusercontent.com/80975570/203384139-4dece104-4215-45d2-8a79-0b8ac1e193d2.png)
With Model Gap Fix:
![2022-11-22_17 39 35](https://user-images.githubusercontent.com/80975570/203384164-5a17f589-7495-4a07-af32-ce4952721737.png)

**Comment by Julienraptor01** [comment]
> The 1.4.1 or 1.4.2 update of Iris seems to have fixed all of the rendering issues that I was having, relating to [this issue](https://github.com/MehVahdJukaar/modelfix-multi/issues/2) (mainly with Continuity builtins and CITs in combination for some reason), and from my experience it does improve the look of some resourcepacks, especially with 3D ones:
> Vanilla:
> ![2022-11-22_17 35 00](https://user-images.githubusercontent.com/80975570/203384139-4dece104-4215-45d2-8a79-0b8ac1e193d2.png)
> With Model Gap Fix:
> ![2022-11-22_17 39 35](https://user-images.githubusercontent.com/80975570/203384164-5a17f589-7495-4a07-af32-ce4952721737.png)
> 

That's normal, the cause of the issue was an optimisation made by the dev of continuity (which contribute to iris too) that had unintended side-effects

**Comment by Thanos1716** [comment]
Yes, glad its fixed, I'll now be using this mod, personally :D

**Comment by Julienraptor01** [comment]
it work perfectly by default, is better because it also fix blocks and even seem to improve performance by a small bit
only downside is that there isn't a config GUI if for some people editing the config was needed
